krystex/src/main/java/com/flipkart/krystal/krystex/node/Node.java

```diff
@@ -267,32 +298,46 @@ private boolean execute(RequestId requestId, ImmutableSet<String> newInputNames)
     return executeMainLogic;
   }
 
-  private void executeResolver(RequestId requestId, ResolverDefinition resolverDefinition) {
+  private void executeResolver(
+      RequestId requestId, ResolverDefinition resolverDefinition, boolean fromSkipDependency) {
     Map<NodeLogicId, ResolverCommand> nodeResults =
         this.resolverResults.computeIfAbsent(requestId, r -> new LinkedHashMap<>());
     String dependencyName = resolverDefinition.dependencyName();
     NodeId depNodeId = nodeDefinition.dependencyNodes().get(dependencyName);
-    Inputs inputsForResolver = getInputsForResolver(resolverDefinition, requestId);
+    Inputs inputsForResolver =
+        getInputsForResolver(resolverDefinition, requestId, fromSkipDependency);
     NodeLogicId nodeLogicId = resolverDefinition.resolverNodeLogicId();
-    ResolverCommand resolverCommand =
-        nodeDefinition
-            .nodeDefinitionRegistry()
-            .logicDefinitionRegistry()
-            .getResolver(nodeLogicId)
-            .resolve(inputsForResolver);
+    ResolverCommand resolverCommand;
+    if (fromSkipDependency) {
+      resolverCommand = ResolverCommand.skip("skipped as parent skipped");
+    } else {
+      resolverCommand =
+          nodeDefinition
+              .nodeDefinitionRegistry()
+              .logicDefinitionRegistry()
+              .getResolver(nodeLogicId)
+              .resolve(inputsForResolver);
+    }
     nodeResults.put(nodeLogicId, resolverCommand);
     DependencyNodeExecutions dependencyNodeExecutions =
         dependencyExecutions
             .computeIfAbsent(requestId, k -> new LinkedHashMap<>())
             .computeIfAbsent(dependencyName, k -> new DependencyNodeExecutions());
     dependencyNodeExecutions.executedResolvers().add(resolverDefinition);
     if (resolverCommand instanceof SkipDependency) {
+      ImmutableList<Inputs> inputs = resolverCommand.getInputs();
+      dependencyNodeExecutions.skippedDependency().putIfAbsent(requestId, true);
```

Comment:
Here we are putting requestId in the skippedDependency map, but while reading we are reading by dependency name, this won't work, no?